components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,343 @@
+            } catch (error) {
+                setCreateError(error);
+            } finally {
+                setCreating(false);
+            }
+        },
+        [repo, selectedIde, selectedWsClass, useLatestIde],
+    );
+
+    const onClickCreate = useCallback(() => createWorkspace(), [createWorkspace]);
+
+    if (SelectAccountPayload.is(selectAccountError)) {
+        return (
+            <SelectAccountModal
+                {...selectAccountError}
+                close={() => {
+                    window.location.href = gitpodHostUrl.asAccessControl().toString();
+                }}
+            />
+        );
+    }
+    if (createResult?.workspaceURL) {
+        window.location.href = createResult.workspaceURL;
+        return <></>;
+    } else if (createResult?.createdWorkspaceId) {
+        return <Redirect to={`/start/${createResult.createdWorkspaceId}`} />;
+    } else if (createResult?.existingWorkspaces) {
+        return <ExistingWorkspaceModal result={createResult} createWorkspace={createWorkspace} onClose={() => {}} />;
+    }
+
+    return (
+        <div className="flex flex-col mt-32 mx-auto ">
+            <div className="flex flex-col max-h-screen max-w-lg mx-auto items-center w-full">
+                <h1>New Workspace</h1>
+                <div className="text-sm text-gray-500">Start a new workspace with the following options.</div>
+                <div className="-mx-6 px-6 mt-6 w-full">
+                    <div className="pt-3">
+                        <RepositoryFinder setSelection={setRepo} initialValue={repo} />
+                    </div>
+                    <div className="pt-3">
+                        {errorIde && <div className="text-red-500 text-sm">{errorIde}</div>}
+                        <SelectIDEComponent
+                            onSelectionChange={onSelectEditorChange}
+                            setError={setErrorIde}
+                            selectedIdeOption={selectedIde}
+                            useLatest={useLatestIde}
+                        />
+                    </div>
+                    <div className="pt-3">
+                        {errorWsClass && <div className="text-red-500 text-sm">{errorWsClass}</div>}
+                        <SelectWorkspaceClassComponent
+                            onSelectionChange={setSelectedWsClass}
+                            setError={setErrorWsClass}
+                            selectedWorkspaceClass={selectedWsClass}
+                        />
+                    </div>
+                </div>
+                <div className="w-full flex justify-end mt-6 space-x-2 px-6">
+                    {creating && (
+                        <div className="flex-1 flex px-3 my-auto">
+                            <img className="h-4 w-4 animate-spin" src={Spinner} alt="loading spinner" />
+                            <span className="typewriter text-sm mx-3 text-gray-500">Creating workspace... </span>
```

Comment:
**question**: Also, is it better to move the loading state out the button instead of showing the button (already) disabled with a spinner and changing the label to creating workspace?

| BEFORE | AFTER :a: | AFTER :b: |
|-|-|-|
| <img width="532" alt="new-before" src="https://user-images.githubusercontent.com/120486/219350059-4b01f203-19d9-4a06-a063-a02f7f709852.png"> | <img width="532" alt="new-after-1" src="https://user-images.githubusercontent.com/120486/219350064-733401db-0981-4377-94ef-02b8e530a055.png"> | <img width="532" alt="new-after-2" src="https://user-images.githubusercontent.com/120486/219350070-672ab149-08a5-4461-bab2-6418310b1a27.png"> |